presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PayloadJoinOptimizer.java

```diff
@@ -527,6 +589,7 @@ private static class JoinContext
         private Map<VariableReferenceExpression, RowExpression> projectionsToPush = new HashMap<>();
 
         int numJoins;
+        List<VariableReferenceExpression> payloadColumns = new ArrayList<>();
```

Comment:
Nit: make all class variables private and consider initializing them to immutable collections